### ISSUE TITLE
Uncaught TypeError: Cannot read property 'focus' of undefined

### ISSUE THREAD

**Issue by jimmont** [vaadin/vaadin-select#125]
duplicate of https://github.com/vaadin/vaadin-list-box/issues/30 (intended to file it here under vaadin-dropdown-menu and mistakenly elsewhere looking through active issues)

when using the following versions and menu from the markup (which appears to work fine) the menu does not appear as a result of this error.

is it more appropriate to file this under `vaadin-list-mixin` or `vaadin-dropdown-menu`?

```
Uncaught TypeError: Cannot read property 'focus' of undefined
    at HTMLElement.focus (vaadin-list-mixin.js:185)
    at HTMLElement._openedChanged (vaadin-dropdown-menu.js:362)
    at Object.runObserverEffect [as fn] (property-effects.js:214)
    at runEffectsForProperty (property-effects.js:159)
    at runEffects (property-effects.js:125)
    at HTMLElement._propertiesChanged (property-effects.js:1707)
    at HTMLElement._flushProperties (properties-changed.js:335)
    at HTMLElement._flushProperties (property-effects.js:1555)
    at HTMLElement._invalidateProperties (property-effects.js:1527)
    at HTMLElement._setProperty (property-effects.js:1512)

return html`...excerpt...
		<vaadin-dropdown-menu placeholder="modules" on-value-changed="appSelect">
		<template>
		<vaadin-list-box>
			${ repeat(this.itemList, (i, index) => index, (i, index) => html`<vaadin-item value="${i.name}">${i.title}</vaadin-item>`) }
		</vaadin-list-box>
		</template>
		</vaadin-dropdown-menu>




npm ls --depth=0
├── @polymer/app-layout@3.0.0-pre.19
├── @polymer/lit-element@0.5.1
├── @vaadin/vaadin-button@2.0.0-pre.4
├── @vaadin/vaadin-development-mode-detector@1.0.3-pre.6
├── @vaadin/vaadin-dropdown-menu@1.0.0-pre.3
├── @vaadin/vaadin-grid@5.0.0-pre.4
├── @vaadin/vaadin-list-box@1.0.1-pre.2
├── @vaadin/vaadin-lumo-styles@1.0.0-pre.6
├── @vaadin/vaadin-tabs@2.0.0-pre.3
├── @webcomponents/webcomponentsjs@2.0.0
```

**Comment by jimmont** [comment]
this reproduces the problem: 2 dropdown menus are created, the one generated with the repeat has the error above when the menu is activated/clicked
```html
<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
<title>vaadin menu from repeat</title>
<style> html, body{background-color:#fff;font-family:sans-serif;} </style>
</head><body>
	<a-menu></a-menu>
<script type="module">
import { LitElement, html } from '/node_modules/@polymer/lit-element/lit-element.js';
import { repeat } from '/node_modules/lit-html/lib/repeat.js';
import '/node_modules/@vaadin/vaadin-dropdown-menu/vaadin-dropdown-menu.js';

class AMenu extends LitElement{
	_render(){
return html`
<style> :host { display:block; } </style>

		
<vaadin-dropdown-menu label="Label" placeholder="Placeholder" value="Option one" on-value-changed="appSelect">
<template>
	<vaadin-list-box>
		<vaadin-item>Option one</vaadin-item>
		<vaadin-item>Option two</vaadin-item>
		<vaadin-item>Option three</vaadin-item>
		<hr>
		<vaadin-item disabled>Option four</vaadin-item>
	</vaadin-list-box>
</template>
</vaadin-dropdown-menu>

<vaadin-dropdown-menu label="Problem on select" placeholder="repeat(itemList)" on-value-changed="appSelect">
<template>
	<vaadin-list-box>
		${ repeat(this.itemList,
			(i, index) => index,
			(i, index) => 
			html`<vaadin-item >${i.title}</vaadin-item>`
		) }
	</vaadin-list-box>
</template>
</vaadin-dropdown-menu>
`;
	}
	appSelect(e={}, detail){
		console.log(e.type, detail, e);
	}
	constructor(){
		super();

		this.itemList = [
			{name: 'okay', title: 'O.K.'}
			,{name: 'stuff', title: 'Stuff + Things'}
			,{name: 'last', title: 'Lastly'}
		];
	}
	static get properties() {
		return {
			itemList: Array
		};
	}
}
window.customElements.define('a-menu', AMenu);
</script>
</body></html>


```

**Comment by web-padawan** [comment]
Thanks for the reproduction. It looks like the `observer.flush()` invocation [here](https://github.com/vaadin/vaadin-list-mixin/blob/master/vaadin-list-mixin.html#L186) does not force `repeat` to render, as it does for the `dom-repeat`. Needs further investigation, though.

**Comment by jimmont** [comment]
using a `dom-repeat` has the same behavior as the `repeat` in my example, I added the import and replaced the section with that element with the same error.

**Comment by cdchow** [comment]
I'm encountering this same issue, curious about when an appropriate fix will be rolled out.

**Comment by YuriyVaadin** [comment]
It seems that issue is similar to those we have with all of the frameworks at the moment (regarding templates). Probably we will be able to provide a proper solution after https://github.com/vaadin/components-team-tasks/issues/372 .
At the moment even this example is not going to render properly:
```
<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
<title>Repeat in the template</title>
<style> html, body{background-color:#fff;font-family:sans-serif;} </style>
</head>
<body>
	<a-menu></a-menu>
	<script type="module">
		import { LitElement, html } from './node_modules/@polymer/lit-element/lit-element.js';
		import { repeat } from './node_modules/lit-html/lib/repeat.js';

		class AMenu extends LitElement{
			_render() {
				return html`
				${ repeat(this.itemList,
					(i, index) => index,
					(i, index) =>
					html`${i.title}`
				) }

				<template>
					${ repeat(this.itemList,
						(i, index) => index,
						(i, index) =>
						html`${i.title}`
					) }
				</template>
				`;
			}

			constructor() {
				super();

				this.itemList = [
					{name: 'okay', title: 'O.K.'}
					,{name: 'stuff', title: 'Stuff + Things'}
					,{name: 'last', title: 'Lastly'}
				];
			}

			static get properties() {
				return {
					itemList: Array
				};
			}
		}
		window.customElements.define('a-menu', AMenu);
	</script>
</body>
</html>
```

**Comment by cdchow** [comment]
Hi @YuriyVaadin, thanks for the update. You linked a private Github issue. Is there anyway to inspect the status of that issue? In my case, it feels rather frustrating to use all of the Lumo themed Vaadin UI components except for the dropdown.

**Comment by YuriyVaadin** [comment]
Hi @cdchow,
Sorry, I am not able to provide the access to track that issue, but I will notify you on updates. Currently we are working on prototyping of the possible solution.

**Comment by tomivirkki** [comment]
Closing, the discussed `renderer` feature was shipped with version 1.2.0
